File: codss.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""Pograma LAMB para códigos de salud de BPS"""
import datos
import htm
import pagina
import cgi
def listado():
    """ Listado de códigos de seguro de salud """
    pag = pagina.Pagina("Listado de códigos del Seguro de Salud", 4)
    htm.button("Nuevo","codss.py?accion=nuevo")
    htm.button('Volver','geined.py?accion=sistema')
    # cargar datos
    codigoss = datos.Tabla("codigoss")
    i = 0
    htm.encabezado_tabla(["Nº", "Categoria", "Acciones"])
    for fila in codigoss.resultado:
        htm.fila_alterna(i)
        ident = fila["id"]
        print(htm.td(ident))
        print(htm.td(fila["codigo"]))
        print("<td>")
        htm.boton_detalles("codss.py?accion=ver&id=" + str(ident))
        htm.boton_editar("codss.py?accion=editar&id=" + str(ident))
        htm.boton_eliminar("codss.py?accion=eliminar&id=" + str(ident))
        print("</td>")
        i = i + 1
    htm.fin_tabla()
    htm.button('Volver','geined.py?accion=sistema')
    pag.fin()
def nuevo():
    """ Formulario para nuevo código de Seguro de Salud """
    pag = pagina.Pagina('Nuevo código de Seguro de Salud')
    htm.form_edicion("Nuevo código de Seguro de Salud", "codss.py?accion=agregar")
    htm.input_texto("Código:", "codigo", "")
    htm.botones('codss.py?accion=listado')
    htm.form_edicion_fin()
    pag.fin
def agregar(frm):
    """Agrega registro a la base de datos codigoss"""
    # Voy por acá
    codigoss = datos.Tabla("codigoss")
    codigoss.buscar("codigo", frm["codigo"])
    if codigoss.num_filas > 0:
        pagina.duplicado("codss.py")
    else:
        codigoss.registro["codigo"] = frm["codigo"]
        codigoss.agregar()
        htm.redirigir("cem.php?accion=listado")
def editar(frm):
    """ Edición de código de Seguro de Salud"""
    pag = pagina.Pagina("Edición de código de seguro de salud", 2)
    htm.encabezado_tabla(["Campo","Valor"])
    htm.formulario("codss.php?accion=actualizar")
    codigoss = datos.Tabla("codigoss")
    codigoss.ir_a(frm["id"])
    print(htm.hidden("id", frm['id']))
    htm.input_texto("Código:", "codigo", codigoss.registro['codigo'])
    htm.botones()
    htm.fin_formulario()
    htm.fin_tabla()
    print(htm.boton('Volver','codss.php?accion=listado'))
    pag.fin()
def actualizar(frm):
    """ Actualiza registro de Código de Seguro de Salud """
    ident = frm['id']
    codigoss = datos.Tabla("codigoss")
    codigoss.ir_a(ident)
    codigoss.registro["codigo"] = frm["codigo"]
    codigoss.actualizar()
    htm.redirigir('codss.php?accion=listado')
def detalle(frm):
    """ Listado de empleados por código de Seguro de Salud"""
    pag = pagina.Pagina("Detalles de CSS de un empleado", 5)
    codigoss = datos.Tabla("codigoss")
    codigoss.ir_a(frm["id"])
    # cli_ver
    print(htm.h2(codigoss.registro['codigo']))
    htm.encabezado_tabla(["Nombre","CI","Direccion","Telefono","eMail","Ingreso","Notas","Acciones"])
    empleados = datos.Tabla("empleados")
    empleados.buscar("codigoss", frm["id"])
    i = 0
    for fil2 in empleados.resultado:
        htm.fila_alterna(i)
        print(htm.td(fil2['nombre']))
        print(htm.td(fil2['ci']))
        print(htm.td(fil2['direccion']))
        print(htm.td(fil2['telefono']))
        print(htm.td(fil2['email']))
        print(htm.td(fil2['ingreso']))
        print(htm.td(fil2['notas']))
        print('<td>')
        print(htm.boton("Detalles", 'emp_ver.php?id=' + str(fil2['id'])))
        print('</tr>')
        i = i + 1
    htm.fin_tabla()
    htm.boton('Volver','codss.php?accion=listado')
    pag.fin()
def main():
    """Principal"""
    form = cgi.FieldStorage()
    accion = form.getvalue("accion","listado")
    if accion == 'listado':
        listado()
    elif accion == "nuevo":
        nuevo()
    elif accion == "agregar":
        agregar(form)
    elif accion == "editar":
        editar(form)
    elif accion == "actulizar":
        actualizar(form)
    elif accion == 'eliminar':
        # TODO borrar('codigoss',$_POST['id'],'codss.php');
        pass
    elif accion == 'ver':
        detalle(form)

if __name__ == "__main__":
    main()
    

File: codss.php
<?php
/*! @file
    Programa LAMB para códigos de salud de BPS
*/
include('funciones.php');
include('datos.php');
function listado(){
    /*! Listado de códigos de seguro de salud
    @param ninguno
    @return Código HTML para listado de seguro de salud
    */
    encabezado('Listado de códigos del Seguro de Salud');
    boton("Nuevo","codss.php?accion=nuevo");
    boton('Volver','geined.py?accion=sistema');
    //cargar datos
    $sql='SELECT * FROM codigoss';
    $resultado = mysql_query($sql);
    $i=0;
    encabezado_tabla(array("Nº","Categoria","Acciones"));
    while($row = mysql_fetch_array($resultado)){
        fila_alterna($i);
        $id = $row['id'];
        celda($id);
        celda($row['codigo']);
        echo '<td>';
        boton("Detalles","codss.php?accion=ver&id=$id");
        boton("Editar","codss.php?accion=editar&id=$id");
        boton("Borrar","codss.php?accion=confirmar&id=$id");
        echo '</td>';
        $i = $i + 1;
    }
    fin_tabla();
    boton('Volver','geined.py?accion=sistema');
    fin();
}
function nuevo(){
    /*! Formulario para nuevo código de Seguro de Salud
    @param ninguno
    @return Código HTML 
    */
    autorizacion(2);
    encabezado('Nuevo código de Seguro de Salud');
    encabezado_tabla(array("Campo","Valor"));
    formulario("codss.php?accion=agregar");
    input_texto("Código:","codigo","");
    botones();
    fin_formulario();
    fin_tabla();
    boton('Volver','codss.php?accion=listado');
}
function agregar(){
    /*! Agrega registro a la base de datos codigoss
    @param $_POST
    @return redirige a listado 
    */
    $sql1 = 'SELECT * FROM codigoss WHERE codigo="'.$_POST['codigo'].'"';
    $res1 = mysql_query($sql1);
    if (mysql_num_rows($res1)==0){
        $sql = 'INSERT INTO codigoss SET codigo = "' . $_POST['codigo'].'"';
        $result = mysql_query($sql) or die('Consulta inválida: ' . mysql_error());
        redirigir("cem.php?accion=listado");
    }
    else {
        duplicado('codss.php?accion=listado');
    }
}
function editar(){
    /*! Edición de código de Seguro de Salud
    @param $_GET['id']
    @return Código HTML 
    */
    autorizacion(2);
    encabezado('Edición de Código de Seguro de Salud');
    encabezado_tabla(array("Campo","Valor"));
    formulario("codss.php?accion=actualizar");
    $sql='SELECT * FROM codigoss WHERE id="' . $_GET['id'].'"';
    //cargar datos
    $resultado = mysql_query($sql);
    $fila=mysql_fetch_array($resultado);
    campo_oculto("id",$_GET['id']);
    input_texto("Código:","codigo",$fila['codigo']);
    botones();
    fin_formulario();
    fin_tabla();
    boton('Volver','codss.php?accion=listado');
}
function actualizar(){
    /*! Actualiza registro de Código de Seguro de Salud
    @param $_POST
    @return SQL, redirige a Listado 
    */
    $id = $_POST['id'];
    $sql = 'UPDATE codigoss SET codigo= "' . $_POST['codigo'] . 
        '" WHERE id = "' . $_POST['id'] . '"';
    $result = mysql_query($sql) or die('Invalid query: ' . mysql_error());
    redirigir('codss.php?accion=listado');
}
function detalle(){
    /*! Listado de empleados por código de Seguro de Salud
    @param $_GET['id']
    @return Código HTML 
    */
    autorizacion(5);
    encabezado('Detalles de Código de Seguro de Salud');
    $sql1 = 'SELECT * FROM codigoss WHERE id = "' . $_GET['id'] . '"';
    $res1 = mysql_query($sql1);
    $fil1 = mysql_fetch_array($res1);
    // cli_ver
    echo '<h2>'.$fil1['codigo'].'</h2>';
    encabezado_tabla(array("Nombre","CI","Direccion","Telefono","eMail","Ingreso","Notas","Acciones"));
    $sql2 = 'SELECT * FROM empleados WHERE codigoss="'.$_GET['id'].'"';
    $res2 = mysql_query($sql2);
    $i = 0;
    while($fil2=mysql_fetch_array($res2)){
        fila_alterna($i);
        Celda($fil2['nombre']);
        Celda($fil2['ci']);
        Celda($fil2['direccion']);
        Celda($fil2['telefono']);
        Celda($fil2['email']);
        Celda($fil2['ingreso']);
        Celda($fil2['notas']);
        echo '<td>';
        boton("Detalles",'emp_ver.php?id='.$fil['id']);
        echo '</tr>';
        $i = $i + 1;
    }
    fin_tabla();
    boton('Volver','codss.php?accion=listado'); 
}
autorizacion(3);
if (!$_GET['accion']) $accion = 'listado';
else $accion=$_GET['accion'];
switch ($accion) {
    case 'listado':
        listado();
        break;
    case 'nuevo':
        nuevo();
        break;
    case 'agregar':
        agregar();
        break;
    case 'editar':
        editar();
        break;
    case 'actualizar':
        actualizar();
        break;
    case 'confirmar':
        confirmar_borrar($_GET['id'],'cem.php');
        break;
    case 'eliminar':
        autorizacion(2);
        borrar('codigoss',$_POST['id'],'codss.php');
        break;
    case 'ver':
        detalle();
        break;
}
?>
